aviator/aviator.go

```
@@ -229,22 +230,41 @@ func (a *Aviator) resolveEnvironment(serviceName string) error {
 	return a.environment.Resolve()
 }
 
-// promptForSetup prompts a user to choose which setup to use for a service.
-func (a *Aviator) promptForSetup(serv *Service) ServiceSetup {
+// promptOrUseConfigForSetup prompts a user to choose which setup to use for a service, or uses a config string, if provided.
+func (a *Aviator) promptOrUseConfigForSetup(serv *Service, useConfig string, input io.Reader) ServiceSetup {
 	setupOptions, defaultConfig := getSetupOptions(*serv)
-	if len(setupOptions) > 1 {
-		log.Normalf("Configuration options for '%s': \n", serv.Name)
-		for _, val := range setupOptions {
-			log.Normalln(val)
+	if useConfig == "" {
+		if len(setupOptions) > 1 {
+			log.Normalf("Configuration options for '%s': \n", serv.Name)
+			for _, val := range setupOptions {
+				log.Normalln(val)
+			}
+			log.Normalf("Config for %s? ", serv.Name)
+			fmt.Fscanf(input, "%s", &serv.selectedSetup)
+			log.Normalln("")
 		}
-		log.Normalf("Config for %s? ", serv.Name)
-		fmt.Scanf("%s", &serv.selectedSetup)
-		log.Normalln("")
+	} else if useConfig != "default" {
+		// "default" falls through to the next case
+		// n.b. This means that if there's a config called "default", that is not itself the default,
+		// it won't get picked. But if someone does that, they diserve a debugging headache.
+		serv.selectedSetup = useConfig
 	}
 	if serv.selectedSetup == "" {
 		serv.selectedSetup = defaultConfig
 	}
 
+	// Ensure the chosen config name is valid.
+	isRealConfig := false
+	for name, _ := range serv.Setups {
+		if name == serv.selectedSetup {
+			isRealConfig = true
+		}
+	}
+	if isRealConfig == false {
+		log.Normalf("Unknown config: %s for service: %s", serv.selectedSetup, serv.Name)
+		// TODO(bstein): Throw an error here?
```

I would throw an error here, to be strict and clear.

Originally when we were making Aviator, we mainly thought of configs as corresponding to exactly one value. e.g. `staging` might mean `http://staging-api.example.com`. So we also thought it would be useful to be able to input something custom at runtime, i.e. literally the user would type `http://my-api.example.com` or `localhost:5000`. I don't believe this has been a commonly used or valuable feature.